pkg/watch/watch.go

```diff
@@ -242,16 +242,23 @@ func (o *WatchClient) WatchAndPush(out io.Writer, parameters WatchParameters, ct
 		return fmt.Errorf("error watching deployment: %v", err)
 	}
 
-	// TODO(feloy) watch files included by Devfile
 	devfileWatcher, err := fsnotify.NewWatcher()
 	if err != nil {
 		return err
 	}
 	if parameters.WatchFiles {
-		err = devfileWatcher.Add(parameters.DevfilePath)
+		var devfileFiles []string
+		devfileFiles, err = libdevfile.GetReferencedLocalFiles(parameters.InitialDevfileObj)
 		if err != nil {
 			return err
 		}
+		devfileFiles = append(devfileFiles, parameters.DevfilePath)
+		for _, f := range devfileFiles {
+			err = devfileWatcher.Add(f)
+			if err != nil {
+				klog.V(4).Infof("error adding watcher for path %s: %v", f, err)
```

Comment:
Shouldn't we return the error right away instead? Or maybe make the error more visible to the user by logging it as a warning?

Reply:
I first started with returning the error, but lots of integration tests fail, because they are using Devfiles containing URIs but withtout the corresponding file. 

I would prefer the error to be handled by the devfile validation. I consider that if the devfile validation does not complain, we should not here.